cdci_data_analysis/analysis/renku_helper.py

```diff
@@ -30,22 +31,18 @@ def push_api_code(api_code,
         step = 'assigning branch name'
         branch_name = get_branch_name(job_id=job_id)
 
-        step = f'checking the branch already exists'
-        job_id_branch_already_exists = check_job_id_branch_is_present(repo, job_id)
+        step = f'checkout branch {branch_name}'
+        repo = checkout_branch_renku_repo(repo, branch_name)
 
-        if not job_id_branch_already_exists:
-            step = f'checkout branch {branch_name}'
-            repo = checkout_branch_renku_repo(repo, branch_name)
+        step = f'removing token from the api_code'
+        token_pattern = r"(\'|\")token(\'|\"):.\s?(\'|\").*?(\'|\"),?"
+        api_code = re.sub(token_pattern, '"token": "<INSERT_YOUR_TOKEN_HERE>",', api_code, flags=re.DOTALL)
```

Comment:
after trying this, I find it would be better if this line is commented-out in the API code. That way the code could be run directly (if allowed)

Reply:
Like so? [https://renkulab.io/gitlab/gabriele.barni/test-dispatcher-endpoint/-/blob/mmoda_request_9a5e3646291b26da/api_code_9a5e3646291b26da.ipynb](https://renkulab.io/gitlab/gabriele.barni/test-dispatcher-endpoint/-/blob/mmoda_request_9a5e3646291b26da/api_code_9a5e3646291b26da.ipynb)